lib/storage-service.js

```diff
@@ -65,6 +65,9 @@ class StorageService {
         // Queue the promise states with a timeout
         return onceWithTimeout(manager, 'ready', this.timeout).then(() => {
           return manager.data;
+        }).catch((err) => {
+          this._managers.delete(manager.name);
```

Comment:
Can we write a test around this to ensure the manager gets deleted when an error's thrown?